ivy_tests/test_ivy/test_frontends/test_numpy/test_manipulation_routines/test_changing_array_shape.py

```diff
@@ -0,0 +1,65 @@
+# global
+import numpy as np
+from hypothesis import given, strategies as st
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+import ivy.functional.backends.numpy as ivy_np
+from ivy_tests.test_ivy.helpers import handle_cmd_line_args
+
+@st.composite
+def _copy_n_with_out(draw):
+    copy = draw(st.booleans())
+    out = draw(st.booleans())
+    return copy, out
+
+@st.composite
+def _x_dtypes_n_shape(draw):
+    min_num_dims = 1
+    max_num_dims = 5
+    min_dim_size = 1
+    max_dim_size = 10
+
+    shape = []
+    shape_len = draw(st.integers(min_num_dims, max_num_dims))
+    for _ in range(shape_len):
+        s_ = draw(st.integers(min_dim_size, max_dim_size))
+        shape.append(s_)
+
+    dtypes, x = draw( helpers.dtype_and_values(shape=tuple(shape)) )
+    reshape = draw(st.permutations(shape))
+    return dtypes, x, tuple(reshape)
```

Comment:
Hi, i don't think these functions are required. You can do the same thing inside the `given` decorator, with the helper functions already provided. Kindly go through these, for example `dtype_and_values`, `array_dtypes` etc.

Reply:
I agree with copy and with out but the second function is used to generate a valid reshape for example: 
a numpy array of shape 3,2 cannot be reshaped to 7,1 so the function _x_dtypes_n_shape would draw an arbitrary shape create an array with the same shape & then shuffles the shape to generate a valid reshape. Alternatively I can randomly draw integers and keep dividing with the product. Is there a default function for that as well.

Reply:
Hi @ojask12, apologies for the late reply. There's a function called `reshape_shapes` in the helper file, please have a look at it, thanks🙂